emission/core/get_database.py

```diff
@@ -21,8 +21,20 @@
 result_limit = config_data["timeseries"]["result_limit"]
 config_file.close()
 
+try:
+    parsed=pymongo.uri_parser.parse_uri(url)
+except:
+    print("URL not formatted, defaulting to \"Stage_database\"")
+    db_name = "Stage_database"
+else:
+    if parsed['database']:
+        db_name = parsed['database']
+    else:
+        print("URL does not specify a DB name, defaulting to \"Stage_database\"")
+        db_name = "Stage_database"
+
```

Comment:
@aGuttman This looks like it will work, but there are also additional locations where we use `Stage_database`
https://github.com/e-mission/e-mission-docs/issues/721#issuecomment-1122684668

Did you have a chance to go through at least the important ones (primarily `abstract_timeseries.py`) to ensure that they don't need to be fixed?

Reply:
`abstract_timeseries.py` includes "Stage_database" in `_reset_url()`, a function which seems to never be used:
```
grep -r -l "_reset_url" .
./emission/storage/timeseries/abstract_timeseries.py
./emission/storage/timeseries/__pycache__/abstract_timeseries.cpython-37.pyc
./htmlcov/d_9b9568736b288f26_abstract_timeseries_py.html
```
`_reset_url()` is defined in `abstract_timeseries.py` but never called inside of it. The other two are artifacts of some kind, from compiling and from coverage testing.

For the db_auth/TestMongodbAuth, these are stand alone tests not run in the unit tests that are hardcoded to localhost. I can update those as well to use the config url, but it seemed unnecessary to worry about them at the time, since the unit tests were passing.

Probably should have asked about these at the time. Is it worth it to update db_auth/TestMongodbAuth? Does not make sense to you that _reset_url() is never called, or am I missing something wrt it?

Reply:
@aGuttman as you can see in the comment, I added _reset_url() as a hack to allow us to connect to multiple databases from a single client instance, so that we could perform analyses over multiple instances at the same time - see https://github.com/e-mission/e-mission-eval-private-data/blob/d9c8f3fa6d2d6c307e38edcfa4e123f4cb8a96ac/tour_model_eval/Federating%20and%20saving%20multiple%20datasets.ipynb

It turns out that we can just `mongorestore` all of them into the same DB instance. So this should not be needed.

At the same time, some level of federation, so that we can have a public dashboard that spans the various CEO locations, for example, would be good. But we should explore the standard mongodb federation functionality (if any) first.

So I am open to removing this code, as long as you file an issue to explore federation over the long term.

Reply:
wrt `db_auth.py` and `TestMongodbAuth.py`, I thought you could modify them to support and test the new solution. 

@aGuttman We don't test DB auth as part of the unit tests, since we don't want to cause dev mongodb installs to become unreachable if the tests fail, but we should test them nevertheless.

Reply:
Ok. I will open a new issue, remove the old code and update the tests.